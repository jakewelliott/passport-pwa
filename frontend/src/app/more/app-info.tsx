import { PassportHeader } from '../../components/common/passport-header';
import { ContactUs } from '../../components/tabs/more/contact-us';
import { DownloadParkMaps } from '../../components/tabs/more/download-park-maps';
import { Reservations } from '../../components/tabs/more/reservations';

export const AppInfo = () => {
  return (
    <div className='max-w-[500px] mx-auto'>
      <div className='my-3.5'>
        <PassportHeader />
      </div>
      <ContactUs />
      <DownloadParkMaps />
      <Reservations />
    </div>
  );
};
