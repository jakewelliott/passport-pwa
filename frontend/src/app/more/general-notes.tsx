import { NotesMiniTab } from '../locations/components/notes-minitab';

export const EditGeneralNotes = () => {
	return (
		<div className='container mx-auto'>
			<div className='flex h-[calc(100vh-1.5rem-100px)] flex-col'>
				<NotesMiniTab parkId={0} />
			</div>
		</div>
	);
};
