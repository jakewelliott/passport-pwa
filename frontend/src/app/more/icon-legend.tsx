import { LegendSections } from '@/app/more/components/legend-sections';
import { dbg } from '@/lib/debug';

export const IconLegend = () => {
    dbg('RENDER', '/more/icon-legend');
    return (
        <div className='flex flex-col gap-4'>
            <h2 className='text-center'>Legend</h2>
            <LegendSections />
            <p className='min-w-[200px]'>
                <span className='text-icon-amenities'>Kids TRACK Trail: </span>Kids in Parks is a group that creates
                outdoor adventures called TRACK trails for families all over the country. Each one has fun things to
                learn about nature as you walk. You can even earn prizes by keeping track of your adventures! Check out
                kidsinparks.com to find out more.
            </p>
        </div>
    );
};
