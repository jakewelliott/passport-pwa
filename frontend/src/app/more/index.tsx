import RoundedButton from "../../components/common/rounded-button";

export default function More() {
	return (
		<>
			<h1>More</h1>
			<RoundedButton title="TEST" color="system_black" />
		</>
	);
}
