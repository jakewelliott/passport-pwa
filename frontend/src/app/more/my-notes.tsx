import { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import ListRow from '../../components/common/list-row';
import { a11yOnClick } from '@/lib/a11y';

export const MyNotes = () => {
  const [generalNotes, setGeneralNotes] = useState('');
  const navigate = useNavigate();
  const parks = [
    {
      name: 'Sample Data',
      abbreviation: 'ABCD',
      address: [
        {
          name: 'Main Address:',
          addressLineOne: '1234 Main St',
          city: 'Raleigh',
          state: 'NC',
          zip: '27606',
        },
        {
          name: 'Secondary Address:',
          addressLineOne: '1234 Main St',
          city: 'Raleigh',
          state: 'NC',
          zip: '27606',
        },
      ],
      coordinates: '35.2023, -78.9761',
      phone: '(555) 555-5555',
      email: 'email@ncparks.gov',
      website: 'ncparks.gov',
      stamp: {
        time: '1/1/25',
        method: 'manual',
      },
      bucketList: {
        text: 'random bucket list item',
        status: false,
      },
      established: '2003',
      landmark: 'My House',
      youCanFind: 'My items',
      trails: 'The driveway',
      parkIcons: ['Paddling-Red.svg', 'RVCamping-Green.svg', 'Playground-Blue.svg'],
      parkPhotos: [
        { url: './photos/CABE.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
      ],
      parkNotes: '',
    },
    {
      name: 'Sample Data',
      abbreviation: 'ABCD',
      address: [
        {
          name: 'Main Address:',
          addressLineOne: '1234 Main St',
          city: 'Raleigh',
          state: 'NC',
          zip: '27606',
        },
        {
          name: 'Secondary Address:',
          addressLineOne: '1234 Main St',
          city: 'Raleigh',
          state: 'NC',
          zip: '27606',
        },
      ],
      coordinates: '35.2023, -78.9761',
      phone: '(555) 555-5555',
      email: 'email@ncparks.gov',
      website: 'ncparks.gov',
      stamp: {
        time: '1/1/25',
        method: 'manual',
      },
      bucketList: {
        text: 'random bucket list item',
        status: false,
      },
      established: '2003',
      landmark: 'My House',
      youCanFind: 'My items',
      trails: 'The driveway',
      parkIcons: ['Paddling-Red.svg', 'RVCamping-Green.svg', 'Playground-Blue.svg'],
      parkPhotos: [
        { url: './photos/CABE.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
      ],
      parkNotes: '',
    },
    {
      name: 'Sample Data',
      abbreviation: 'ABCD',
      address: [
        {
          name: 'Main Address:',
          addressLineOne: '1234 Main St',
          city: 'Raleigh',
          state: 'NC',
          zip: '27606',
        },
        {
          name: 'Secondary Address:',
          addressLineOne: '1234 Main St',
          city: 'Raleigh',
          state: 'NC',
          zip: '27606',
        },
      ],
      coordinates: '35.2023, -78.9761',
      phone: '(555) 555-5555',
      email: 'email@ncparks.gov',
      website: 'ncparks.gov',
      stamp: {
        time: '1/1/25',
        method: 'manual',
      },
      bucketList: {
        text: 'random bucket list item',
        status: false,
      },
      established: '2003',
      landmark: 'My House',
      youCanFind: 'My items',
      trails: 'The driveway',
      parkIcons: ['Paddling-Red.svg', 'RVCamping-Green.svg', 'Playground-Blue.svg'],
      parkPhotos: [
        { url: './photos/CABE.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
      ],
      parkNotes: 'heyyo',
    },
    {
      name: 'Sample Data',
      abbreviation: 'ABCD',
      address: [
        {
          name: 'Main Address:',
          addressLineOne: '1234 Main St',
          city: 'Raleigh',
          state: 'NC',
          zip: '27606',
        },
        {
          name: 'Secondary Address:',
          addressLineOne: '1234 Main St',
          city: 'Raleigh',
          state: 'NC',
          zip: '27606',
        },
      ],
      coordinates: '35.2023, -78.9761',
      phone: '(555) 555-5555',
      email: 'email@ncparks.gov',
      website: 'ncparks.gov',
      stamp: {
        time: '1/1/25',
        method: 'manual',
      },
      bucketList: {
        text: 'random bucket list item',
        status: false,
      },
      established: '2003',
      landmark: 'My House',
      youCanFind: 'My items',
      trails: 'The driveway',
      parkIcons: ['Paddling-Red.svg', 'RVCamping-Green.svg', 'Playground-Blue.svg'],
      parkPhotos: [
        { url: './photos/CABE.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
        { url: './photos/CACR.jpg' },
      ],
      parkNotes:
        ' this is some super lajnfiawlrbjhvkjg hcfjtgfkgvkjhvkyufvvvvvvvv vvvvvvvvvvvvvvvvvvvvvv vvvvvvvvvvvvvvvvvvvvvvvvvvvvvv h hvhvhvhvhvhvhvhvhvhvhvhvhvhvhvhvhv hvhvhvhvhvhvhvhvhvhvhvhvhvhvhvhvhvhvh vhvhvhvhvhvhvhvhvhvhvhvhvhvhv hvhvhvhvhvhvhvhvhvhvhvhhvhv',
    },
  ];
  const parksWithNotes = parks.filter((park) => park.parkNotes && park.parkNotes.trim() !== '');

  useEffect(() => {
    const savedNotes = localStorage.getItem('generalNotes');
    if (savedNotes) {
      setGeneralNotes(savedNotes);
    }
  }, []);

  return (
    <div className='container mx-auto px-4 py-4'>
      <div className='space-y-4'>
        <div {...a11yOnClick(() => navigate('/more/my-notes/general-notes'))} className='cursor-pointer'>
          <ListRow>
            <div className='flex flex-col gap-1'>
              <h3>General Notes</h3>
              <p className='mb-2 text-gray-500 text-sm'>Last updated: Not available</p>
              <p
                className='overflow-wrap-anywhere line-clamp-3 max-w-full hyphens-auto break-words'
                style={{ hyphens: 'auto' }}
              >
                {generalNotes || 'No general notes yet'}
              </p>
            </div>
          </ListRow>
        </div>

        {/* Park Notes section */}
        {parksWithNotes.length === 0 ? (
          <p className='text-gray-600'>No park notes found.</p>
        ) : (
          parksWithNotes.map((park, index) => (
            <div
              key={index}
              {...a11yOnClick(() => navigate(`/locations/location-detail/${park.abbreviation}?tab=notes`))}
              className='cursor-pointer'
            >
              <ListRow>
                <div className='flex flex-col gap-1'>
                  <h3>{park.name}</h3>
                  <p>{park.address[0].city}</p>
                  <p className='mb-2 text-gray-500 text-sm'>Last updated: Not available</p>
                  <p
                    className='overflow-wrap-anywhere line-clamp-3 max-w-full hyphens-auto break-words'
                    style={{ hyphens: 'auto' }}
                  >
                    {park.parkNotes}
                  </p>
                </div>
              </ListRow>
            </div>
          ))
        )}
      </div>
    </div>
  );
};
