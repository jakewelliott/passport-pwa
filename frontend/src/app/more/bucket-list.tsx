import { BucketListItem } from '@/components/tabs/more/bucket-list-item';

export const BucketList = () => {
  return (
    <div className='p-7'>
      <div className='max-w-96 mx-auto m-7'>
        <BucketListItem />
        <BucketListItem />
        <BucketListItem />
        <BucketListItem />
        <BucketListItem />
        <BucketListItem />
        <BucketListItem />
        <BucketListItem />
        <BucketListItem />
        <BucketListItem />
        <BucketListItem />
      </div>
    </div >
  );
};
