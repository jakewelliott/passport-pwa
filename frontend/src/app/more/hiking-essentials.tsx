import { HikingEssentialItem } from '@/components/tabs/more/hiking-essential-item';

export const HikingEssentials = () => {
  return (
    <div className='mx-auto max-w-4xl p-6' lang='en'>
      <h3 className='mb-5 bg-supporting_inactiveblue p-3 text-center text-system_white uppercase'>
        WHAT TO PACK FOR A HIKE
      </h3>

      <div className='flex flex-wrap justify-center gap-6'>
        <HikingEssentialItem
          imageSrc='/icons/hiking/boot.svg'
          altText='Hiking boot'
          title='HIKING BOOTS'
          description='Make sure your boots are broken in before hitting the trails to prevent blisters and sore feet!'
        />

        <HikingEssentialItem
          imageSrc='/icons/hiking/water-bottle.svg'
          altText='Water bottle'
          title='WATER'
          description="You need to drink about 1 liter of water per every 4-5 miles hiked. Don't forget water for your pet!"
        />

        <HikingEssentialItem
          imageSrc='/icons/hiking/map-and-compass.svg'
          altText='Map and compass'
          title='MAP & COMPASS'
          description="Pick up a park map from the park's visitor center or download a digital version at ncparks.gov"
        />

        <HikingEssentialItem
          imageSrc='/icons/hiking/snacks.svg'
          altText='Snacks'
          title='SNACKS'
          description='You can make your own trail mix with dark chocolate, almonds, raisins and unsalted popcorn!'
        />

        <HikingEssentialItem
          imageSrc='/icons/hiking/rain-gear.svg'
          altText='Rain gear'
          title='RAIN GEAR'
          description='When hiking in wet weather, avoid wearing cotton - it traps water and takes a long time to dry.'
        />

        <HikingEssentialItem
          imageSrc='/icons/hiking/flashlight.svg'
          altText='Flashlight'
          title='FLASHLIGHT'
          description='Due to all the trees, the trails will get darker before the rest of the park.'
        />

        <HikingEssentialItem
          imageSrc='/icons/hiking/first-aid-kit.svg'
          altText='First aid kit'
          title='FIRST AID KIT'
          description='Make sure your kit has bandages, antiseptic, antihistamines, and tweezers.'
        />

        <HikingEssentialItem
          imageSrc='/icons/hiking/matches.svg'
          altText='Matches'
          title='MATCHES'
          description='Pack waterproof matches or carry them in a waterproof container.'
        />

        <HikingEssentialItem
          imageSrc='/icons/hiking/knife.svg'
          altText='Knife'
          title='KNIFE'
          description='Make sure your knife or multitool has a locking blade to prevent any injuries.'
        />
      </div>
    </div>
  );
};
