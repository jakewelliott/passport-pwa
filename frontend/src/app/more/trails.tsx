import { TrailDetails } from '@/components/tabs/more/trail-details';
import { TrailMap } from '@/components/tabs/more/trail-map';

export const Trails = () => {
  const trailDetails: TrailDetails[] = [
    {
      trailName: 'Dan River State Trail',
      trailIcons: ['Hiking'],
      distance: '90 miles',
      description: 'northern Piedmont Triad',
    },
    {
      trailName: 'Deep River State Trail',
      trailIcons: [],
      distance: '125 miles',
      description: 'Piedmont',
    },
    {
      trailName: 'East Coast Greenway State Trail',
      trailIcons: [],
      distance: '795 miles',
      description:
        'V-shaped: one arm through Raleigh, Durham and the Triangle and other arm through Albemarle and Pamlico Sounds, both meeting at Wilmington',
    },
    {
      trailName: 'Equine State Trail',
      trailIcons: [],
      distance: '350 miles',
      description: 'southern Piedmont',
    },
    {
      trailName: 'Fonta Flora State Trail',
      trailIcons: [],
      distance: '100 miles',
      description: 'Asheville and the Foothills / Lake James',
    },
    {
      trailName: 'French Broad River State Trail',
      trailIcons: [],
      distance: '117 miles',
      description: 'western border',
    },
    {
      trailName: 'Haw River State Trail',
      trailIcons: [],
      distance: '80 miles',
      description: 'Piedmont Triad / west of the Triangle',
    },
    {
      trailName: 'Hickory Nut Gorge State Trail',
      trailIcons: [],
      distance: '50 miles',
      description: 'southwest of Asheville',
    },
    {
      trailName: 'Mountains-to-Sea State Trail',
      trailIcons: [],
      distance: '1,400-miles',
      description: 'from Great Smoky Mountains National Park to the Outer Banks',
    },
    {
      trailName: 'Northern Peaks State Trail',
      trailIcons: [],
      distance: '40 miles',
      description: 'High Country',
    },
    {
      trailName: 'Overmountain Victory State Trail',
      trailIcons: [],
      distance: '225 miles',
      description: 'west',
    },
    {
      trailName: 'Roanoke River State Trail',
      trailIcons: [],
      distance: '132 miles',
      description: 'Albemarle Sound',
    },
    {
      trailName: 'Wilderness Gateway State Trail',
      trailIcons: [],
      distance: '170 miles',
      description: 'southwest of Asheville / the Foothills',
    },
    {
      trailName: 'Yadkin River State Trail',
      trailIcons: [],
      distance: '162 miles',
      description: 'Piedmont',
    },
  ];

  return (
    <div className='m-6'>
      {trailDetails.map((trail, index) => (
        <TrailDetails trail={trail} key={index} />
      ))}
      <img src='/TrailsLogo.svg' />
      <p className='text-center'>
        For maps and additional information on these state trails, please visit{' '}
        <a href='https://trails.nc.gov/state-trails' className='text-main_blue'>
          trails.nc.gov/state-trails
        </a>
      </p>
      <TrailMap />
    </div>
  );
};
