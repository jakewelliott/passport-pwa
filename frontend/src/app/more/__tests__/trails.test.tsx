import { Trails } from '@/app/more/trails';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { describe, expect, it } from 'vitest';
describe('Trails Component - User Stories', () => {
    // TODO: fix - test doesn't match actual component content
    // // User Story: As a user, I want to learn about trail difficulty levels
    // it('should display information about trail difficulty levels', () => {
    // 	render(
    // 		<BrowserRouter>
    // 			<Trails />
    // 		</BrowserRouter>
    // 	);

    // 	expect(screen.getByText(/Easy/)).toBeInTheDocument();
    // 	expect(screen.getByText(/Moderate/)).toBeInTheDocument();
    // 	expect(screen.getByText(/Strenuous/)).toBeInTheDocument();
    // });

    // User Story: As a user, I want to understand trail ratings
    it('should explain trail ratings', () => {
        render(
            <BrowserRouter>
                <Trails />
            </BrowserRouter>,
        );

        expect(screen.getByText(/Dan River State Trail/i)).toBeInTheDocument();
        expect(screen.getByText(/Deep River State Trail/i)).toBeInTheDocument();
        expect(screen.getByText(/East Coast Greenway State Trail/i)).toBeInTheDocument();
    });

    it('should explain trail difficulty', () => {
        render(
            <BrowserRouter>
                <Trails />
            </BrowserRouter>,
        );

        expect(screen.getByText(/90 miles/i)).toBeInTheDocument();
        expect(screen.getByText(/125 miles/i)).toBeInTheDocument();
        expect(screen.getByText(/795 miles/i)).toBeInTheDocument();
    });

    // User Story: As a user, I want to learn about trail safety
    it('should provide trail safety information', () => {
        render(
            <BrowserRouter>
                <Trails />
            </BrowserRouter>,
        );

        expect(screen.getByText(/For maps and additional information/i)).toBeInTheDocument();
        expect(screen.getByText(/trails.nc.gov\/state-trails/i)).toBeInTheDocument();
    });

    // User Story: As a user, I want to understand trail accessibility
    it('should display accessibility information', () => {
        render(
            <BrowserRouter>
                <Trails />
            </BrowserRouter>,
        );

        expect(screen.getByText(/Dan River State Trail/i)).toBeInTheDocument();
        expect(screen.getByText(/90 miles/i)).toBeInTheDocument();
        expect(screen.getByText(/northern Piedmont Triad/i)).toBeInTheDocument();
    });

    // User Story: As a user, I want to learn about trail etiquette
    it('should provide trail etiquette guidelines', () => {
        render(
            <BrowserRouter>
                <Trails />
            </BrowserRouter>,
        );

        expect(screen.getByText(/For maps and additional information/i)).toBeInTheDocument();
        expect(screen.getByText(/trails.nc.gov\/state-trails/i)).toBeInTheDocument();
    });
});
