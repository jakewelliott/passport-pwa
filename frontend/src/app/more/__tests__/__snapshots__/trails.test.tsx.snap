// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Trails Component - User Stories > matches snapshot 1`] = `
<div>
  <div
    class="animate-pulse"
    data-testid="loading-placeholder"
  >
    <p
      class="text-center text-gray-500 text-sm"
    >
       Loading trails...
    </p>
  </div>
</div>
`;
